### DISCUSSION TITLE
Trouble with Encoding of Versioning

### DISCUSSION THREAD

**Discussion by nicolaischneider** [outfoxx/PotentCodables#8] [Q&A] [answered]
Hi,

I'm having some issues with the versioning scheme. How can I encode the following structure in asn1: `Version ::= INTEGER { v1(1) }`, in other words, how should structure and scheme look like?

Thanks for your help in advance,
Nico

**Comment by kdubb** [comment, marked as answer]
Here's how `TBSCertficate`'s schema is specified, which starts with a version: 

https://github.com/outfoxx/Shield/blob/a885653d1555624080e6f0e66e2229825d386f23/Sources/ShieldX509/TBSCertificate.swift#L103
